snow/engine/snowman/bootstrap/bootstrapper_test.go

```diff
@@ -1487,62 +710,111 @@ func TestBootstrapperReceiveStaleAncestorsMessage(t *testing.T) {
 	)
 	require.NoError(err)
 
-	vm.CantSetState = false
 	require.NoError(bs.Start(context.Background(), 0))
 
-	vm.GetBlockF = func(_ context.Context, blkID ids.ID) (snowman.Block, error) {
-		switch blkID {
-		case blkID0:
-			return blk0, nil
-		case blkID1:
-			if blk1.StatusV == choices.Accepted {
-				return blk1, nil
-			}
-			return nil, database.ErrNotFound
-		case blkID2:
-			if blk2.StatusV == choices.Accepted {
-				return blk2, nil
-			}
-			return nil, database.ErrNotFound
-		default:
-			require.FailNow(database.ErrNotFound.Error())
-			return nil, database.ErrNotFound
-		}
-	}
-	vm.ParseBlockF = func(_ context.Context, blkBytes []byte) (snowman.Block, error) {
-		switch {
-		case bytes.Equal(blkBytes, blkBytes0):
-			return blk0, nil
-		case bytes.Equal(blkBytes, blkBytes1):
-			return blk1, nil
-		case bytes.Equal(blkBytes, blkBytes2):
-			return blk2, nil
-		default:
-			require.FailNow(errUnknownBlock.Error())
-			return nil, errUnknownBlock
-		}
-	}
-
 	requestIDs := map[ids.ID]uint32{}
-	sender.SendGetAncestorsF = func(_ context.Context, vdr ids.NodeID, reqID uint32, blkID ids.ID) {
-		require.Equal(peerID, vdr)
+	sender.SendGetAncestorsF = func(_ context.Context, nodeID ids.NodeID, reqID uint32, blkID ids.ID) {
+		require.Equal(peerID, nodeID)
 		requestIDs[blkID] = reqID
 	}
 
-	require.NoError(bs.startSyncing(context.Background(), []ids.ID{blkID1, blkID2})) // should request blk2 and blk1
+	require.NoError(bs.startSyncing(context.Background(), blocksToIDs(blks[1:3]))) // should request blk1 and blk2
 
-	reqIDBlk1, ok := requestIDs[blkID1]
+	reqIDBlk1, ok := requestIDs[blks[1].ID()]
 	require.True(ok)
-	reqIDBlk2, ok := requestIDs[blkID2]
+	reqIDBlk2, ok := requestIDs[blks[2].ID()]
 	require.True(ok)
 
-	require.NoError(bs.Ancestors(context.Background(), peerID, reqIDBlk2, [][]byte{blkBytes2, blkBytes1}))
-
+	require.NoError(bs.Ancestors(context.Background(), peerID, reqIDBlk2, blocksToBytes(blks[1:3])))
 	require.Equal(snow.Bootstrapping, config.Ctx.State.Get().State)
-	require.Equal(choices.Accepted, blk0.Status())
-	require.Equal(choices.Accepted, blk1.Status())
-	require.Equal(choices.Accepted, blk2.Status())
+	requireStatusIs(require, blks, choices.Accepted)
 
-	require.NoError(bs.Ancestors(context.Background(), peerID, reqIDBlk1, [][]byte{blkBytes1}))
+	require.NoError(bs.Ancestors(context.Background(), peerID, reqIDBlk1, blocksToBytes(blks[1:2])))
 	require.Equal(snow.Bootstrapping, config.Ctx.State.Get().State)
 }
+
+func generateBlockchain(length uint64) []snowman.Block {
```

Comment:
kinda wild to me that we didn't have this already... It might make sense for some of these functions to be move into a common testing package (because they seem useful in a number of places)